frontend/src/components/templates/Stats.js

```diff
@@ -74,18 +74,50 @@ export const Stats = () => {
   const [stats, setStats] = useState(null);
   const { targetApi } = useParams();
 
+  async function getTargetAPIList(targetApi) {
+    let result = [];
+
+    if (targetApi === 'allApi') {
+      const ApiTargetConfiguration = pickBy(
+        DATA_PROVIDER_PARAMETERS,
+        (dataProviderConfig) => dataProviderConfig.type === 'api'
+      );
+      const apiList = Object.keys(ApiTargetConfiguration);
+      result = await getAPIStats(apiList);
+    } else if (targetApi === 'allServices') {
+      const serviceTargetConfiguration = pickBy(
+        DATA_PROVIDER_PARAMETERS,
+        (dataProviderConfig) => dataProviderConfig.type === 'service'
+      );
+      const serviceList = Object.keys(serviceTargetConfiguration);
+      result = await getAPIStats(serviceList);
+    } else if (targetApi === undefined) {
+      result = await getAPIStats([]);
+    } else {
+      result = await getAPIStats([targetApi]);
+    }
+    return result;
+  }
+
   useEffect(() => {
     async function fetchStats() {
-      const result = await getAPIStats(targetApi);
+      let result = [];
+      try {
+        result = await getTargetAPIList(targetApi);
 
-      setStats({
-        ...result.data,
-        enrollment_by_target_api: stackLowUseAndUnpublishedApi(
-          DATA_PROVIDER_WITH_ENROLLMENTS_IN_PRODUCTION_ENV,
-          result.data.enrollment_by_target_api,
-          10
-        ),
-      });
+        setStats({
+          ...result.data,
+          enrollment_by_target_api: stackLowUseAndUnpublishedApi(
+            DATA_PROVIDER_WITH_ENROLLMENTS_IN_PRODUCTION_ENV,
+            result.data.enrollment_by_target_api,
+            10
+          ),
+        });
+      } catch (error) {
+        console.error(
```

Comment:
there can be other errors here, do not catch error this way (we can do that in a PP session)